In [7]:
import os
import numpy as np
from PIL import Image
import pandas as pd
import matplotlib as mat
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
path = './input/data/train/'

In [3]:
df = pd.read_csv(path+'train.csv')
df.head()

id  gender   race  age                    path
0  000001  female  Asian   45  000001_female_Asian_45
1  000002  female  Asian   52  000002_female_Asian_52
2  000004    male  Asian   54    000004_male_Asian_54
3  000005  female  Asian   58  000005_female_Asian_58
4  000006  female  Asian   59  000006_female_Asian_59

In [4]:
df_race = df['race']
df_race.unique()

array(['Asian'], dtype=object)

데이터의 인종은 'Asian'으로 유일하다

In [5]:
print('min age : {}, max age: {}'.format(df['age'].min(), df['age'].max()))

print('나이별 인원수')
for first in range(10,70,10):
    filtered = (df['age'] >= first) & (df['age'] < (first+10))
    print(first,'대 : ',len(df[filtered]))


min age : 18, max age: 60
나이별 인원수
10 대 :  599
20 대 :  682
30 대 :  95
40 대 :  192
50 대 :  940
60 대 :  192


나이 10대와 50대의 데이터가 편향되어 있어 Data Augmentation이 필요한 것으로 보인다

In [6]:
image_paths = os.path.join(path,'images')

file_dir = os.listdir(image_paths)
file_dir = [_ for _ in file_dir if '._' not in _]

temp_file_dir = os.listdir(os.path.join(image_paths,file_dir[0]))
temp_file_dir = [_ for _ in temp_file_dir if '._' not in _]

image_file_path = os.path.join(os.path.join(image_paths,file_dir[0]),temp_file_dir[0])

im = Image.open(image_file_path)

print(im)



<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=384x512 at 0x7FB46C63F730>


모든 이미지 파일은 (384, 512)의 크기와 RGB의 3채널의 정보를 가지고 있다.

In [34]:
df = pd.read_csv(path+'/data.csv')
classes = np.sort(df['class'].unique())

count = [0 for _ in range(18)]

df_label = df['class']
df_label.value_counts(sort=True)


4.0     4085
3.0     3660
0.0     2745
1.0     2050
16.0     817
10.0     817
15.0     732
9.0      732
6.0      549
12.0     549
5.0      545
2.0      415
13.0     410
7.0      410
17.0     109
11.0     109
14.0      83
8.0       83
Name: class, dtype: int64

클래스 0, 3, 4가 압도적으로 많아서 학습 시 편향에 주의해야 할 듯합니다